### PR TITLE
Add GraphQl Filed Duplication KB Entry

### PR DESCRIPTION
**Pull request by ybadaoui-ostorlab**
This PR is for adding The GraphQl Filed Duplication KB Entry

### CODE REVIEW

**Review by nmasdoufi-ol — APPROVED** [review]

File: WEB_SERVICE/WEB/_HIGH/FIELD_DUPLICATION/recommendation.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+              while self.peek(graphql.TokenKind.NAME):
+                  field_name = self.parse_field().name.value
+                  field_counts[field_name] = field_counts.get(field_name, 0) + 1
+                  if field_counts[field_name] > api.API_MAX_FIELD_DUPLICATES:
+                      raise graphql.GraphQLError("Exception - Max field duplicates exceeded")
+              return []
+  
+      ast_parser = FieldDuplicationParser(query)
+      ast_parser.parse_document()
+```
```
Comment (ohachimOs) [inline WEB_SERVICE/WEB/_HIGH/FIELD_DUPLICATION/recommendation.md:56]: Did you check the preview for this?

File: WEB_SERVICE/WEB/_HIGH/FIELD_DUPLICATION/recommendation.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+To mitigate the risk of Field Duplication attacks, you can take the following steps:
+
+- **Implement Query Complexity Limits:** Enforce query complexity rules that consider duplicated fields as part of the overall cost of a query. This helps in limiting the number of duplicate fields processed, thus protecting the server from resource exhaustion.
+- **Limit Field Repetitions:** Configure server-side limits on the number of times a field can be duplicated in a single GraphQL query. You can use tools like GraphQL Armor to enforce such limits and prevent field duplication overloading.
+
+=== "JavaScript"
+
+```
```
Comment (rabsondev) [inline WEB_SERVICE/WEB/_HIGH/FIELD_DUPLICATION/recommendation.md:8]:
```suggestion
```javascript
```

File: WEB_SERVICE/WEB/_HIGH/FIELD_DUPLICATION/recommendation.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+      // Propagate rejection details to the client | default: true
+      propagateOnRejection: true,
+    }
+  })
+```
+
+=== "Python"
+
+```
```
Comment (rabsondev) [inline WEB_SERVICE/WEB/_HIGH/FIELD_DUPLICATION/recommendation.md:32]:
```suggestion
```python
```

**Review by rabsondev — DISMISSED** [review]

File: WEB_SERVICE/WEB/_HIGH/FIELD_DUPLICATION/recommendation.md (hunk 1 above)
Comment (ybadaoui-ostorlab) [reply WEB_SERVICE/WEB/_HIGH/FIELD_DUPLICATION/recommendation.md]: fixed, thank you!

**Review by ohachimOs — APPROVED** [review]

File: WEB_SERVICE/WEB/_HIGH/FIELD_DUPLICATION/recommendation.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+=== "JavaScript"
+  ```JavaScript
+   // Configuring for GraphQL Armor
+   GraphQLArmorConfig({
+           maxFieldDuplicates: {
+                   // Enable or disable the plugin | default: true
+               enabled: true,
+
+      // Set the maximum number of field duplications allowed per query | default: 10
```
Comment (ErebusZ) [inline WEB_SERVICE/WEB/_HIGH/FIELD_DUPLICATION/recommendation.md:14]:
```suggestion
           maxFieldDuplicates: {
      // Enable or disable the plugin | default: true
               enabled: true,

      // Set the maximum number of field duplications allowed per query | default: 10
```

File: WEB_SERVICE/WEB/_HIGH/FIELD_DUPLICATION/meta.json [hunk 5]
```diff
@@ -0,0 +1,37 @@
+{
+    "risk_rating": "high",
+    "title": "Field Duplication in GraphQL API",
+    "short_description": "Exploiting a GraphQL server by duplicating fields, potentially leading to Denial of Service attacks.",
+    "references": {
+      "imperva": "https://www.imperva.com/blog/graphql-vulnerabilities-common-attacks/",
+      "escape": "https://docs.escape.tech/testing/vulnerabilities/resource_limitation/graphql_field_duplication",
```
Comment (3asm) [inline WEB_SERVICE/WEB/_HIGH/FIELD_DUPLICATION/meta.json:7]: Please avoid both.

Comment (3asm) [reply WEB_SERVICE/WEB/_HIGH/FIELD_DUPLICATION/meta.json:7]: Prefer resources by OWASP and non-company orgs.

File: WEB_SERVICE/WEB/_HIGH/FIELD_DUPLICATION/meta.json [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+      "CWE_TOP_25": [
+        "CWE_400"
+      ],
+      "PCI_STANDARDS": [
+        "REQ_6_2",
+        "REQ_6_4",
+        "REQ_11_3"
+      ],
+      "OWASP_MASVS_L2": [
+        "MSTG_PLATFORM_2"
```
Comment (3asm) [inline WEB_SERVICE/WEB/_HIGH/FIELD_DUPLICATION/meta.json:22]: I think other MASVS are missing, L2 seems to high for this?

**Review by 3asm — CHANGES_REQUESTED** [review]

File: WEB_SERVICE/WEB/_HIGH/FIELD_DUPLICATION/meta.json (hunk 5 above)
Comment (ybadaoui-ostorlab) [reply WEB_SERVICE/WEB/_HIGH/FIELD_DUPLICATION/meta.json:7]: fixed!

File: WEB_SERVICE/WEB/_HIGH/FIELD_DUPLICATION/meta.json (hunk 6 above)
Comment (ybadaoui-ostorlab) [reply WEB_SERVICE/WEB/_HIGH/FIELD_DUPLICATION/meta.json:22]: i did it based on other KBs that are previously merged

File: .gitignore [hunk 7]
```diff
@@ -1,11 +1,166 @@
...
-#
-# If you need to keep large files in Bitbucket, consider introducing
-# Git Large File Storage (Git LFS) as part of your workflow.
-#
-# Learn more about .gitignore:
-#     https://www.atlassian.com/git/tutorials/saving-changes/gitignore
-# Learn more about Git LFS for large files:
-#     https://www.atlassian.com/git/tutorials/git-lfs
\ No newline at end of file
+# Byte-compiled / optimized / DLL files
```
Comment (3asm) [inline .gitignore:1]: Why are you adding this? This has nothing to do with the PR?

File: WEB_SERVICE/WEB/_HIGH/FIELD_DUPLICATION/meta.json [hunk 8]
```diff
@@ -0,0 +1,53 @@
+{
+    "risk_rating": "high",
```
Comment (3asm) [inline WEB_SERVICE/WEB/_HIGH/FIELD_DUPLICATION/meta.json:2]: medium

**Review by 3asm — CHANGES_REQUESTED** [review]

File: WEB_SERVICE/WEB/_HIGH/FIELD_DUPLICATION/meta.json (hunk 8 above)
Comment (ybadaoui-ostorlab) [reply WEB_SERVICE/WEB/_HIGH/FIELD_DUPLICATION/meta.json:2]: noted

Comment (ybadaoui-ostorlab) [reply WEB_SERVICE/WEB/_HIGH/FIELD_DUPLICATION/meta.json]: fixed

File: .gitignore (hunk 7 above)
Comment (ybadaoui-ostorlab) [reply .gitignore]: fixed

**Review by 3asm — APPROVED** [review]

**Pull request merged**